bin/download

```diff
@@ -0,0 +1,279 @@
+        creature.keyword(:double_strike)
+      when /\Aindestructible/i
+        creature.keyword(:indestructable)
+      when /\Atrample( \(this creature can deal excess combat damage to the player or planeswalker it's attacking\.\))?/i
+        creature.keyword(:trample)
+      when /\Aprotection from creatures/i
+        creature.protect("card.creature?")
+      when /\Aprotection from (#{COLORS}) and from (#{COLORS})/i
+        creature.protect("(card.colors & [:#{$1}, :#{$2}]).any?")
+      when /\Aprotection from (#{COLORS})/i
+        creature.protect("card.colors.include?(:#{$1})")
+      when /\Aprotection from all colors/i
+        creature.protect("!card.colorless?")
+      when /\Aprotection from multicolored/i
+        creature.protect("card.multi_colored?")
+      when /\Aprotection from monocolored/i
+        creature.protect("card.colors.length == 1")
+      when /\Aprotection from mana value (\d+) or greater/i
+        creature.protect("card.converted_mana_cost >= #{$1}")
+      when /\Aprotection from (dragons|vampires)/i
+        creature.protect("card.creature? && card.type?(\"#{$1.capitalize[0...-1]}\")")
+      when /\Aprotection from artifacts/i
+        creature.protect("card.artifact?")
+      when /\Aprotection from enchantments/i
+        creature.protect("card.enchantment?")
+      when /\A\{T\}: Destroy target creature token\./
+        creature.ability(<<~DOC)
+          ActivatedAbility.new(
+            mana_cost: {},
+            additional_costs: [Costs::Tap.new(self)],
+            ability: -> { game.add_effect(Effects::Destroy.new(choices: battlefield.creatures.select { |card| card.is_a?(CreatureToken) })) }
+          )
+        DOC
+      when /\A\{T\}: Destroy target tapped creature\./
+        creature.ability(<<~DOC)
+          ActivatedAbility.new(
+            mana_cost: {},
+            additional_costs: [Costs::Tap.new(self)],
+            ability: -> { game.add_effect(Effects::Destroy.new(choices: battlefield.creatures.select(&:tapped?))) }
+          )
+        DOC
+      when /\A\{T\}: Destroy target Wall\./
+        creature.ability(<<~DOC)
+          ActivatedAbility.new(
+            mana_cost: {},
+            additional_costs: [Costs::Tap.new(self)],
+            ability: -> { game.add_effect(Effects::Destroy.new(choices: battlefield.creatures.select { |card| card.type?(\"Wall\") })) }
+          )
+        DOC
+      when /\A\{T\}: Destroy target (#{COLORS}) creature\./
+        creature.ability(<<~DOC)
+          ActivatedAbility.new(
+            mana_cost: {},
+            additional_costs: [Costs::Tap.new(self)],
+            ability: -> { game.add_effect(Effects::Destroy.new(choices: battlefield.creatures.select { |card| card.colors.include?(:#{$1}) })) }
+          )
+        DOC
+      when /\A\{T\}: You gain (\d+) life\./
+        creature.ability(<<~DOC)
+          ActivatedAbility.new(
+            mana_cost: {},
```

Comment:
These blank `mana_cost` makes me think `ActivatedAbility` should be changed to just have a list of costs, rather than a `mana_cost` and `additional_costs`.

Reply:
I've just handled this now here: https://github.com/radar/mtg/commit/079d83473763e6a1e566da446c2b32b142f840bf. You might want to update this script accordingly.